client/js/contextMenuFactory.js

```diff
@@ -0,0 +1,193 @@
+"use strict";
+const $ = require("jquery");
+const socket = require("./socket");
+const utils = require("./utils");
+const JoinChannel = require("./join-channel");
+const ContextMenu = require("./contextMenu");
+const contextMenuActions = [];
+const contextMenuItems = [];
+
+module.exports = {
+	addContextMenuItem,
+	createContextMenu,
+};
+
+addDefaultItems();
+
+/**
+ *
+ * @param opts
+ * @param {function(Object)} [opts.check] - Function to check whether item should show on the context menu, called with the target jquery element, shows if return is truthy
+ * @param {string|function(Object)} opts.className - class name for the menu item, should be prefixed for non-default menu items (if function, called with jquery element, and uses return value)
+ * @param {string|function(Object)} opts.data - data that will be sent to the callback function (if function, called with jquery element, and uses return value)
+ * @param {string|function(Object)} opts.display - text to display on the menu item (if function, called with jquery element, and uses return value)
+ * @param {boolean} [opts.divider] - Whether to put a divider after this option in the menu
```

Comment:
I think it would be nicer to have a dedicated `addDivider` function instead.

That would match previous APIs I have seen so far, but mostly, I think it makes for a slightly simpler usage: when adding items conditionally (`if (...) { addItem(...) }`), you'd require extra check to add a divider after the last item, while an extra function would just go at the end.

If this is unclear, and this might very well be, I'm happy to give a proper example.

Reply:
Yeah, that makes sense. Will do that tomorrow.

Reply:
@astorije updated that.